src/core/qgsprojectviewsettings.h

```diff
@@ -0,0 +1,145 @@
+#ifndef QGSPROJECTVIEWSETTINGS_H
+#define QGSPROJECTVIEWSETTINGS_H
+
+#include "qgis_core.h"
+#include "qgsreferencedgeometry.h"
+#include <QObject>
+#include <QVector>
+
+class QDomElement;
+class QgsReadWriteContext;
+class QDomDocument;
+
+/**
+ * Contains settings and properties relating to how a QgsProject should be displayed inside
+ * map canvas, e.g. map scales and default view extent for the project.
+ *
+ * \ingroup core
+ * \since QGIS 3.10.1
+ */
+class CORE_EXPORT QgsProjectViewSettings : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Constructor for QgsProjectViewSettings with the specified \a parent object.
+     */
+    QgsProjectViewSettings( QObject *parent = nullptr );
+
+    /**
+     * Resets the settings to a default state.
+     */
+    void reset();
+
+    /**
+     * Returns the default view extent, which should be used as the initial map extent
+     * when this project is opened.
+     *
+     * \warning When a project is opened in the QGIS desktop application and saved, individual
+     * map canvases will store their own previous view extent as custom project properties. Reloading
+     * this saved version of the project will trigger the canvases to restore their individual last view
+     * extents. Accordingly, in the QGIS desktop application, this setting only forms a default, initial
+     * view used when the project is opened for the very first time.
+     *
+     * \see setDefaultViewExtent()
+     */
+    QgsReferencedRectangle defaultViewExtent() const;
+
+    /**
+     * Sets the default view \a extent, which should be used as the initial map extent
+     * when this project is opened.
+     *
+     * \warning When a project is opened in the QGIS desktop application and saved, individual
+     * map canvases will store their own previous view extent as custom project properties. Reloading
+     * this saved version of the project will trigger the canvases to restore their individual last view
+     * extents. Accordingly, in the QGIS desktop application, this setting only forms a default, initial
+     * view used when the project is opened for the very first time.
+     *
+     * \see defaultViewExtent()
+     */
+    void setDefaultViewExtent( const QgsReferencedRectangle &extent );
+
+    /**
+     * Sets the list of custom project map \a scales.
+     *
+     * The \a scales list consists of a list of scale denominator values, e.g.
+     * 1000 for a 1:1000 scale.
+     *
+     * \see mapScales()
+     * \see mapScalesChanged()
+     */
+    void setMapScales( const QVector<double> &scales );
+
+    /**
+     * Returns the list of custom project map scales.
+     *
+     * The scales list consists of a list of scale denominator values, e.g.
+     * 1000 for a 1:1000 scale.
+     *
+     * \see setMapScales()
+     * \see mapScalesChanged()
+     */
+    QVector<double> mapScales() const;
+
+    /**
+     * Sets whether project mapScales() are \a enabled.
```

Comment:
Should we be more verbose here? If project scales are enabled, are they added to the QGIS-wide list of predefined scales or are they overwriting those? I assume it's the latter, worth specifying here.

Reply:
resolved